### PR TITLE
Fix bugs in course sync that may result in invalid or duplicate course connection records

### PR DESCRIPTION
**Pull request by weilai-irl**

### CODE REVIEW

File: local/o365/classes/feature/courserequest/main.php [hunk 1]
```diff
@@ -124,18 +124,16 @@ public function get_unmatched_teams_by_user_oid(string $userobjectid) {
...
-        $matchedgroupoids = $DB->get_fieldset_sql($teamsoidsql, array_merge(['group'], $subtypeparams));
+              WHERE type = 'group'
+                AND subtype IN ('teamfromgroup', 'course', 'courseteam')";
+        $matchedgroupoids = $DB->get_fieldset_sql($teamsoidsql);
 
         // Get SDS courses.
-        $sdsmatchedgroupoids = $DB->get_fieldset_select('local_o365_objects', 'objectid', 'type = ? AND subtype = ?',
-            ['sdssection', 'course']);
+        $sdsmatchedgroupoids = $DB->get_fieldset_select('local_o365_objects', 'objectid',
+            'type = "sdssection" AND subtype = "course"');
```
Comment (usqfowlerj) [inline local/o365/classes/feature/courserequest/main.php:136]: These double-quotes aren't valid for PostgreSQL. There's many other instances in this commit too.
```
ERROR:  column "sdssection" does not exist
LINE 1: ...objectid FROM mdl_local_o365_objects WHERE type = "sdssectio...
                                                             ^
```

File: local/o365/classes/page/acp.php [hunk 2]
```diff
@@ -1131,8 +1131,9 @@ public function mode_teamconnections_connect() {
                 $DB->insert_record('local_o365_objects', $grouprecord);
             }
 
-            $DB->delete_records('local_o365_objects', ['type' => 'group', 'subtype' => 'courseteam', 'moodleid' => $courseid]);
-            $DB->delete_records('local_o365_objects', ['type' => 'group', 'subtype' => 'teamfromgroup', 'moodleid' => $courseid]);
+            $DB->delete_records_select('local_o365_objects',
+                'type => "group" AND subtype IN ("courseteam", "teamfromgroup") AND moodleid = ?', [$courseid]);
```
Comment (usqfowlerj) [inline local/o365/classes/page/acp.php:1135]: Presumably the `=>` here was intended to be a `=`. Several other instances in this commit also exist.
```
ERROR:  syntax error at or near "=>"
LINE 1: SELECT * FROM mdl_local_o365_objects WHERE type => "group" A...
                                                        ^
```

File: local/o365/classes/feature/courserequest/main.php (hunk 1 above)
Comment (weilai-irl) [reply local/o365/classes/feature/courserequest/main.php:136]: @usqfowlerj Thank you so much for reporting the issue. We have made a new release containing the fix. Please check it out.

File: local/o365/classes/page/acp.php (hunk 2 above)
Comment (weilai-irl) [reply local/o365/classes/page/acp.php:1135]: @usqfowlerj Thank you so much for reporting the issue. We have made a new release containing the fix. Please check it out.

**Pull request merged**